numpyro/primitives.py

```diff
@@ -131,6 +131,29 @@ def sample(
         + "_unobserved"`` which should be used by guides.
     :return: sample from the stochastic `fn`.
     """
+    if not isinstance(fn, numpyro.distributions.Distribution):
+        type_error = TypeError(
+            "It looks like you tried to use a fn that isn't an instance of "
+            "numpyro.distributions.Distribution, or "
+            "tensorflow_probability.distributions.Distribution. If you're using "
+            "tensorflow_probability, make sure it is correctly installed."
+        )
```

Comment:
Looks like I also need to account for some `Funsor` stuff, which I think is why the tests are failing. Will take a closer look later.

Reply:
Yeah, maybe we can try to import `funsor` first, if success then checks if it is a funsor distribution. Otherwise, following the below logic.